### ISSUE TITLE
SQLSTATE[42S22]: Column not found: 1054 Unknown column 'currency_id'

### ISSUE THREAD

**Issue by overkill32** [monicahq/monica#4756]
**Describe the bug**
When using the UI to export data, I am presented with this error message:
```
Whoops! Something went wrong.

SQLSTATE[42S22]: Column not found: 1054 Unknown column 'currency_id' in 'field list' (SQL: select `id`, `account_id`, `contact_id`, `in_debt`, `status`, `amount`, `currency_id`, `reason`, `created_at`, `updated_at` from `dukedebts` where `account_id` = 1)
```

**Which version are you using:**
Using self-hosted version: 2.19.1 in a docker image.  Database has been through multiple updates using `docker-compose exec app php artisan monica:update`.

**Additional context**
The debts table appears to be missing the currency_id column.
```
mysql> describe dukedebts;
+------------+------------------+------+-----+------------+----------------+
| Field      | Type             | Null | Key | Default    | Extra          |
+------------+------------------+------+-----+------------+----------------+
| id         | int(10) unsigned | NO   | PRI | NULL       | auto_increment |
| account_id | int(10) unsigned | NO   | MUL | NULL       |                |
| contact_id | int(10) unsigned | NO   | MUL | NULL       |                |
| in_debt    | varchar(255)     | NO   |     | no         |                |
| status     | varchar(255)     | NO   |     | inprogress |                |
| amount     | decimal(8,2)     | NO   |     | NULL       |                |
| reason     | longtext         | YES  |     | NULL       |                |
| created_at | timestamp        | YES  |     | NULL       |                |
| updated_at | timestamp        | YES  |     | NULL       |                |
+------------+------------------+------+-----+------------+----------------+
9 rows in set (0.00 sec)
```

The migrations table as follows:
```
mysql> select * from dukemigrations;
+-----+------------------------------------------------------------------------------------+-------+
| id  | migration                                                                          | batch |
+-----+------------------------------------------------------------------------------------+-------+
|   1 | 2014_10_12_000000_create_users_table                                               |     1 |
|   2 | 2014_10_12_100000_create_password_resets_table                                     |     1 |
|   3 | 2016_06_01_000001_create_oauth_auth_codes_table                                    |     1 |
|   4 | 2016_06_01_000002_create_oauth_access_tokens_table                                 |     1 |
|   5 | 2016_06_01_000003_create_oauth_refresh_tokens_table                                |     1 |
|   6 | 2016_06_01_000004_create_oauth_clients_table                                       |     1 |
|   7 | 2016_06_01_000005_create_oauth_personal_access_clients_table                       |     1 |
|   8 | 2016_06_07_234741_create_account_table                                             |     1 |
|   9 | 2016_06_08_003006_add_account_info_table                                           |     1 |
|  10 | 2016_06_08_005413_create_contacts_table                                            |     1 |
|  11 | 2016_06_25_224219_create_reminder_type_table                                       |     1 |
|  12 | 2016_06_28_191025_create_tasks_table                                               |     1 |
|  13 | 2016_06_30_185050_create_notes_table                                               |     1 |
|  14 | 2016_07_25_133835_add_width_field                                                  |     1 |
|  15 | 2016_08_28_122938_create_kids_table                                                |     1 |
|  16 | 2016_08_28_215159_create_relations_table                                           |     1 |
|  17 | 2016_09_03_202027_add_reminder_id_to_contacts                                      |     1 |
|  18 | 2016_09_05_134937_add_last_talked_to_field                                         |     1 |
|  19 | 2016_09_05_135927_add_people_id_to_contacts                                        |     1 |
|  20 | 2016_09_05_145111_add_name_info_to_peoples                                         |     1 |
|  21 | 2016_09_06_213550_create_activity_type_table                                       |     1 |
|  22 | 2016_09_10_164406_create_jobs_table                                                |     1 |
|  23 | 2016_09_10_170122_create_notifications_table                                       |     1 |
|  24 | 2016_09_12_014120_create_failed_jobs_table                                         |     1 |
|  25 | 2016_09_30_014720_add_kid_to_reminder                                              |     1 |
|  26 | 2016_10_15_024156_add_deleted_at_to_users                                          |     1 |
|  27 | 2016_10_19_155139_create_cache_table                                               |     1 |
|  28 | 2016_10_19_155800_create_sessions_table                                            |     1 |
|  29 | 2016_10_21_022941_add_statistics_table                                             |     1 |
|  30 | 2016_10_24_013543_add_journal_setting_to_users                                     |     1 |
|  31 | 2016_10_24_014257_create_journal_tables                                            |     1 |
|  32 | 2016_10_28_002518_add_metric_to_settings                                           |     1 |
|  33 | 2016_11_01_014353_create_activities_table                                          |     1 |
|  34 | 2016_11_01_015957_add_icon_column                                                  |     1 |
|  35 | 2016_11_03_150307_add_activity_location_to_activities                              |     1 |
|  36 | 2016_11_09_013049_add_events_table                                                 |     1 |
|  37 | 2016_12_08_011555_remove_type_from_notes                                           |     1 |
|  38 | 2016_12_13_133945_add_gifts_table                                                  |     1 |
|  39 | 2016_12_28_150831_change_title_column                                              |     1 |
|  40 | 2017_01_14_200815_add_facebook_columns_to_users_table                              |     1 |
|  41 | 2017_01_15_045025_add_colors_to_users                                              |     1 |
|  42 | 2017_01_22_142645_add_fields_to_contacts                                           |     1 |
|  43 | 2017_01_23_043831_change_people_to_contact_for_kids                                |     1 |
|  44 | 2017_01_26_013524_change_people_to_significantother                                |     1 |
|  45 | 2017_01_26_022852_change_notes_to_contact                                          |     1 |
|  46 | 2017_01_26_034553_add_notes_count_to_contact                                       |     1 |
|  47 | 2017_01_27_024356_change_people_in_events                                          |     1 |
|  48 | 2017_01_28_180156_remove_deleted_at_from_significant_others                        |     1 |
|  49 | 2017_01_28_184901_remove_deleted_at_from_kids                                      |     1 |
|  50 | 2017_01_28_193913_remove_deleted_at_from_notes                                     |     1 |
|  51 | 2017_01_28_222114_remove_viewed_at_from_contacts                                   |     1 |
|  52 | 2017_01_29_175146_remove_delete_at_from_activities                                 |     1 |
|  53 | 2017_01_29_175629_add_number_activities_to_contacts                                |     1 |
|  54 | 2017_01_31_025849_add_activity_statistics_table                                    |     1 |
|  55 | 2017_02_02_232450_add_confirmation                                                 |     1 |
|  56 | 2017_02_04_225618_change_reminders_table                                           |     1 |
|  57 | 2017_02_05_035925_add_gifts_metrics_to_contacts                                    |     1 |
|  58 | 2017_02_05_041740_change_gifts_table                                               |     1 |
|  59 | 2017_02_05_042122_change_people_to_contact_for_gifts                               |     1 |
|  60 | 2017_02_07_041607_change_tasks_table                                               |     1 |
|  61 | 2017_02_07_051355_add_number_tasks_to_contact                                      |     1 |
|  62 | 2017_02_08_002251_change_number_tasks_contact                                      |     1 |
|  63 | 2017_02_08_025358_add_sort_preferences_to_users                                    |     1 |
|  64 | 2017_02_10_195613_remove_notifications_table                                       |     1 |
|  65 | 2017_02_10_214714_remove_people_table                                              |     1 |
|  66 | 2017_02_10_215405_remove_entities_table                                            |     1 |
|  67 | 2017_02_10_215705_remove_deleted_at_from_contact                                   |     1 |
|  68 | 2017_02_10_224355_calculate_statistics                                             |     1 |
|  69 | 2017_02_11_154900_add_avatars_to_contacts                                          |     1 |
|  70 | 2017_02_12_134220_create_entries_table                                             |     1 |
|  71 | 2017_05_03_155254_move_significant_other_data                                      |     1 |
|  72 | 2017_05_04_164723_remove_contact_encryption                                        |     1 |
|  73 | 2017_05_04_185921_add_title_to_activities                                          |     1 |
|  74 | 2017_05_04_193252_alter_activity_nullable                                          |     1 |
|  75 | 2017_05_08_164514_remove_encryption_tasks                                          |     1 |
|  76 | 2017_05_30_002239_remove_predefined_reminders                                      |     1 |
|  77 | 2017_05_30_023116_create_money_table                                               |     1 |
|  78 | 2017_06_07_173437_add_multiple_genders_choices                                     |     1 |
|  79 | 2017_06_10_152945_add_social_networks_to_contacts                                  |     1 |
|  80 | 2017_06_10_155349_create_currencies_data                                           |     1 |
|  81 | 2017_06_11_025227_remove_encryption_journal                                        |     1 |
|  82 | 2017_06_11_110735_change_unique_constraint_for_contacts                            |     1 |
|  83 | 2017_06_13_035059_remove_gifts_encryption                                          |     1 |
|  84 | 2017_06_13_195740_add_company_to_contacts                                          |     1 |
|  85 | 2017_06_14_131803_remove_bern_timezone                                             |     1 |
|  86 | 2017_06_14_132911_add_zar_currency_to_currencies_table                             |     1 |
|  87 | 2017_06_16_215256_add_about_who_to_reminders                                       |     1 |
|  88 | 2017_06_17_010900_fix_contacts_table                                               |     1 |
|  89 | 2017_06_17_153814_refactor_user_table                                              |     1 |
|  90 | 2017_06_19_105842_add_stripe_fields_to_users                                       |     1 |
|  91 | 2017_06_20_121345_add_invitations_statistics                                       |     1 |
|  92 | 2017_06_22_210813_add_name_order_to_users                                          |     1 |
|  93 | 2017_06_27_134704_create_import_table                                              |     1 |
|  94 | 2017_06_29_211725_add_import_job_to_statistics                                     |     1 |
|  95 | 2017_06_29_230523_add_gravatar_url_to_users                                        |     1 |
|  96 | 2017_07_02_155736_create_tags_table                                                |     1 |
|  97 | 2017_07_04_132743_add_tags_to_statistics                                           |     1 |
|  98 | 2017_07_09_164312_update_bad_translation_key                                       |     1 |
|  99 | 2017_07_12_014244_create_calls_table                                               |     1 |
| 100 | 2017_07_17_005012_drop_reminders_count_from_contacts                               |     1 |
| 101 | 2017_07_18_215312_add_danish_kroner_to_currencies_table                            |     1 |
| 102 | 2017_07_18_215758_add_indian_rupee_to_currencies_table                             |     1 |
| 103 | 2017_07_19_094503_add_brazilian_real_to_currencies                                 |     1 |
| 104 | 2017_07_22_153209_create_instance_table                                            |     1 |
| 105 | 2017_07_26_220021_change_contacts_table                                            |     1 |
| 106 | 2017_08_02_152838_change_string_to_boolean_for_reminders                           |     1 |
| 107 | 2017_08_06_085629_change_events_data                                               |     1 |
| 108 | 2017_08_06_153253_move_kids_to_contacts                                            |     1 |
| 109 | 2017_08_16_041431_add_contact_avatar_location                                      |     1 |
| 110 | 2017_08_21_224835_remove_paid_limitations_for_current_users                        |     1 |
| 111 | 2017_09_10_125918_remove_unusued_counters                                          |     1 |
| 112 | 2017_09_13_095923_add_tracking_table                                               |     1 |
| 113 | 2017_09_13_191714_add_partial_notion                                               |     1 |
| 114 | 2017_10_14_083556_change_gift_column_structure                                     |     1 |
| 115 | 2017_10_17_170803_change_gift_structure                                            |     1 |
| 116 | 2017_10_19_134816_create_activity_contact_table                                    |     1 |
| 117 | 2017_10_19_135215_move_activities_to_pivot_table                                   |     1 |
| 118 | 2017_10_25_102923_remove_contact_id_activities_table                               |     1 |
| 119 | 2017_11_01_122541_add_met_through_to_contacts                                      |     1 |
| 120 | 2017_11_02_202601_add_is_dead_to_contacts                                          |     1 |
| 121 | 2017_11_10_174654_create_contact_fields_table                                      |     1 |
| 122 | 2017_11_10_181043_migrate_contacts_information                                     |     1 |
| 123 | 2017_11_10_202620_move_addresses_from_contact_to_addresses                         |     1 |
| 124 | 2017_11_10_204035_delete_contact_fields_from_contacts                              |     1 |
| 125 | 2017_11_20_115635_change-amount-to-double-on-debts                                 |     1 |
| 126 | 2017_11_27_083043_add_more_statistics                                              |     1 |
| 127 | 2017_11_27_134403_add_new_avatar_to_contacts                                       |     1 |
| 128 | 2017_11_27_202857_change_tasks_table_structure                                     |     1 |
| 129 | 2017_12_01_113748_update_notes                                                     |     1 |
| 130 | 2017_12_04_164831_create_ages_table                                                |     1 |
| 131 | 2017_12_04_165421_move_ages_data                                                   |     1 |
| 132 | 2017_12_10_181535_remove_important_dates_table                                     |     1 |
| 133 | 2017_12_10_205328_add_account_id_to_activities                                     |     1 |
| 134 | 2017_12_10_214545_add_last_consulted_at_to_contacts                                |     1 |
| 135 | 2017_12_13_115857_create_day_table                                                 |     1 |
| 136 | 2017_12_21_163616_update_journal_entries_with_existing_activities                  |     1 |
| 137 | 2017_12_21_170327_add_google2fa_secret_to_users                                    |     1 |
| 138 | 2017_12_24_115641_create_pets_table                                                |     1 |
| 139 | 2017_12_31_114224_add_dashboard_tab_to_users                                       |     1 |
| 140 | 2018_01_15_105858_create_additional_reminders_table                                |     1 |
| 141 | 2018_01_16_203358_add_gift_received                                                |     1 |
| 142 | 2018_01_16_212320_rename_gift_columns                                              |     1 |
| 143 | 2018_01_17_230820_add_gift_tab_view_to_users                                       |     1 |
| 144 | 2018_01_27_014146_add_custom_gender                                                |     1 |
| 145 | 2018_02_25_202752_change_locale_in_db                                              |     1 |
| 146 | 2018_02_28_223747_update_notification_table                                        |     1 |
| 147 | 2018_03_03_204440_create_relationship_type_table                                   |     1 |
| 148 | 2018_03_18_085815_populate_default_relationship_type_tables                        |     1 |
| 149 | 2018_03_18_090209_populate_relationship_type_tables_with_default_values            |     1 |
| 150 | 2018_03_18_090345_migrate_current_relationship_table_to_new_relationship_structure |     1 |
| 151 | 2018_03_24_083258_migrate_offsprings                                               |     1 |
| 152 | 2018_04_04_220850_create_default_modules_table                                     |     1 |
| 153 | 2018_04_04_222608_create_account_modules_table                                     |     1 |
| 154 | 2018_04_10_205655_fix_production_error                                             |     1 |
| 155 | 2018_04_10_222515_migrate-modules                                                  |     1 |
| 156 | 2018_04_13_131008_fix-contacts-data                                                |     1 |
| 157 | 2018_04_13_205231_create_changes_table                                             |     1 |
| 158 | 2018_04_14_081052_fix_wrong_gender                                                 |     1 |
| 159 | 2018_04_19_190239_stay_in_touch                                                    |     1 |
| 160 | 2018_05_06_061227_external_countries                                               |     1 |
| 161 | 2018_05_06_194710_delete_reminder_sent_table                                       |     1 |
| 162 | 2018_05_07_070458_create_terms_table                                               |     1 |
| 163 | 2018_05_13_110706_add_ex_wife_husband_relationship                                 |     1 |
| 164 | 2018_05_16_143631_add_nickname_to_contacts                                         |     1 |
| 165 | 2018_05_16_214222_add_timestamps_to_currencies                                     |     1 |
| 166 | 2018_05_20_121028_accept_terms                                                     |     1 |
| 167 | 2018_05_20_225034_change_name_order_user-_preferencies                             |     1 |
| 168 | 2018_05_24_160546_fix-inconsistant-reminder-time                                   |     1 |
| 169 | 2018_06_10_191450_add_love_metadata_relationshisp                                  |     1 |
| 170 | 2018_06_10_221746_migrate_entries_objects                                          |     1 |
| 171 | 2018_06_11_184017_change_default_user_table                                        |     1 |
| 172 | 2018_06_13_000100_create_u2f_key_table                                             |     1 |
| 173 | 2018_06_14_212502_change_default_name_order_user_table                             |     1 |
| 174 | 2018_07_03_204220_create_default_activity_type_groups_table                        |     1 |
| 175 | 2018_07_08_104306_update-timestamps-timezone                                       |     1 |
| 176 | 2018_07_26_104306_create-conversations                                             |     1 |
| 177 | 2018_08_06_145046_add_starred_to_contacts                                          |     1 |
| 178 | 2018_08_09_18000_fix-empty-reminder-time                                           |     1 |
| 179 | 2018_08_18_180426_add_legacy_free_plan                                             |     1 |
| 180 | 2018_08_29_124804_add_conversations_to_statistics                                  |     1 |
| 181 | 2018_08_29_222051_add_conversations_to_modules                                     |     1 |
| 182 | 2018_08_31_020908_create_life_events_table                                         |     1 |
| 183 | 2018_09_02_150531_contact_archiving                                                |     1 |
| 184 | 2018_09_05_025008_add_default_profile_view                                         |     1 |
| 185 | 2018_09_05_213507_mark_modules_migrated                                            |     1 |
| 186 | 2018_09_13_135926_add_description_field_to_contacts                                |     1 |
| 187 | 2018_09_18_142844_remove_events                                                    |     1 |
| 188 | 2018_09_23_024528_add_documents_table                                              |     1 |
| 189 | 2018_09_29_114125_add_reminder_to_life_events                                      |     1 |
| 190 | 2018_10_01_211757_add_number_of_views                                              |     1 |
| 191 | 2018_10_04_181116_life_event_vehicle                                               |     1 |
| 192 | 2018_10_07_120133_fix_json_column                                                  |     1 |
| 193 | 2018_10_16_000703_add_documents_to_module_table                                    |     1 |
| 194 | 2018_10_19_081816_life_event_tattoo                                                |     1 |
| 195 | 2018_10_27_230346_fix_non_english_tab_slugs                                        |     1 |
| 196 | 2018_10_28_165814_email_verified                                                   |     1 |
| 197 | 2018_11_11_145035_remove_changelogs_table                                          |     1 |
| 198 | 2018_11_15_172333_make_contact_id_nullable_in_tasks                                |     1 |
| 199 | 2018_11_18_021908_create_images_table                                              |     1 |
| 200 | 2018_11_21_212932_add_contacts_uuid                                                |     1 |
| 201 | 2018_11_25_020818_add_contact_photo_table                                          |     1 |
| 202 | 2018_11_30_154729_recovery_codes                                                   |     1 |
| 203 | 2018_12_08_233140_add_who_called_to_calls                                          |     1 |
| 204 | 2018_12_09_023232_add_emotions_table                                               |     1 |
| 205 | 2018_12_09_145956_create_emotion_call_table                                        |     1 |
| 206 | 2018_12_16_195440_add_gps_coordinates_to_addressess                                |     1 |
| 207 | 2018_12_19_002819_create_places_table                                              |     1 |
| 208 | 2018_12_19_003444_move_addresses_data                                              |     1 |
| 209 | 2018_12_21_235418_add_weather_table                                                |     1 |
| 210 | 2018_12_22_021123_add_weather_preferences_to_users                                 |     1 |
| 211 | 2018_12_22_200413_add_reminder_initial_date_to_reminders                           |     1 |
| 212 | 2018_12_24_164256_add_companies_table                                              |     1 |
| 213 | 2018_12_24_220019_add_occupations_table                                            |     1 |
| 214 | 2018_12_25_001736_add_linkedin_to_default_contact_field_type                       |     1 |
| 215 | 2018_12_25_012011_move_linkedin_data_to_contact_field_type                         |     1 |
| 216 | 2018_12_29_091017_default_temperature_scale                                        |     1 |
| 217 | 2018_12_29_135516_sync_token                                                       |     1 |
| 218 | 2019_01_05_152329_add_reminder_ids_to_contacts                                     |     1 |
| 219 | 2019_01_05_152405_migrate_previous_remiders                                        |     1 |
| 220 | 2019_01_05_152456_drop_special_date_id_from_reminders                              |     1 |
| 221 | 2019_01_05_152526_schedule_new_reminders                                           |     1 |
| 222 | 2019_01_05_202557_add_foreign_keys_to_reminder                                     |     1 |
| 223 | 2019_01_05_202748_add_foreign_key_to_reminder_rule                                 |     1 |
| 224 | 2019_01_05_202938_add_foreign_key_to_contacts                                      |     1 |
| 225 | 2019_01_05_203201_add_foreign_key_for_reminder_in_life-events_table                |     1 |
| 226 | 2019_01_06_135133_update_u2f_key_table                                             |     1 |
| 227 | 2019_01_06_150143_add_inactive_flag_to_reminders                                   |     1 |
| 228 | 2019_01_06_190036_u2f_key_name                                                     |     1 |
| 229 | 2019_01_11_142944_add_foreign_keys_to_activities                                   |     1 |
| 230 | 2019_01_17_093812_add_admin_user                                                   |     1 |
| 231 | 2019_01_18_142032_add_dav_uuid                                                     |     1 |
| 232 | 2019_01_24_221539_change_activity_model_location                                   |     1 |
| 233 | 2019_02_08_234959_remove_users_without_account                                     |     1 |
| 234 | 2019_02_09_200203_add_gender_type                                                  |     1 |
| 235 | 2019_02_17_112452_add_default_gender                                               |     1 |
| 236 | 2019_02_20_205744_allow_gender_null                                                |     1 |
| 237 | 2019_02_24_223855_remove_relation_type_name                                        |     1 |
| 238 | 2019_01_31_223600_add_swiss_chf_to_currencies_table                                |     2 |
| 239 | 2019_03_27_103012_set_default_profile_links                                        |     2 |
| 240 | 2019_03_29_163611_add_webauthn                                                     |     2 |
| 241 | 2019_05_05_194746_add_cron_schedule                                                |     2 |
| 242 | 2019_01_11_183717_change_activities_date_type                                      |     3 |
| 243 | 2019_01_22_034555_create_emotion_activity_table                                    |     3 |
| 244 | 2019_05_15_205533_rename_preferences                                               |     3 |
| 245 | 2019_05_26_000000_add_relationship_table_indexes                                   |     3 |
| 246 | 2019_05_27_000000_populate_relationship_type_tables_with_stepparent_values         |     3 |
| 247 | 2019_08_12_213308_change_avatars_structure                                         |     3 |
| 248 | 2019_08_12_222938_create_avatars_for_existing_contacts                             |     3 |
| 249 | 2019_08_12_222938_create_avatars_for_existing_contacts                             |     4 |
| 250 | 2019_08_13_160332_add_me_contact_on_user                                           |     3 |
| 251 | 2019_08_14_091427_update_stripe_columns                                            |     3 |
| 252 | 2019_09_04_075311_fix_tattoo_or_piercing_translation                               |     3 |
| 253 | 2019_12_17_024553_add_foreign_keys                                                 |     3 |
| 254 | 2019_12_21_100315_change_gift_status                                               |     3 |
| 255 | 2019_12_21_194559_add_photo_gift                                                   |     3 |
| 256 | 2019_12_27_23533_rename_picnicked                                                  |     3 |
| 257 | 2020_02_03_015403_create_audit_log_table                                           |     3 |
+-----+------------------------------------------------------------------------------------+-------+
257 rows in set (0.01 sec)
```

Thank you!!!

**Comment by lpdyck** [comment]
It looks like there are 7 migrations that haven't run on that database, including the one that adds `currency_id` to the `debts` table. Maybe try running `php artisan migrate`?

**Comment by overkill32** [comment]
Thank you for the response.  Results of 'migrate' listed below. It seems to be failing on 2020_02_18_211620_add_contact_field_label.

```
$ docker-compose exec app php artisan migrate
Migrating: 2020_02_18_211620_add_contact_field_label

In Connection.php line 671:
                                                                                                                                                                                                                     
  SQLSTATE[42S01]: Base table or view already exists: 1050 Table 'dukecontact_field_labels' already exists (SQL: create table `dukecontact_field_labels` (`id` bigint unsigned not null auto_increment primary key,  
   `account_id` int unsigned not null, `label_i18n` varchar(20) null, `label` varchar(500) null, `created_at` timestamp null, `updated_at` timestamp null) default character set utf8mb4 collate 'utf8mb4_unicode_c  
  i')                                                                                                                                                                                                                
                                                                                                                                                                                                                     

In PDOStatement.php line 119:
                                                                                                            
  SQLSTATE[42S01]: Base table or view already exists: 1050 Table 'dukecontact_field_labels' already exists  
                                                                                                            

In PDOStatement.php line 117:
                                                                                                            
  SQLSTATE[42S01]: Base table or view already exists: 1050 Table 'dukecontact_field_labels' already exists

**Comment by asbiin** [comment]
Thanks you @lpdyck to have pointed the problem.
@overkill32 you must run all migration step, if the previous migration has failed, you should either
- restore your database backup, and try again
- fix it manually, depending on the specific message you can get
For these 2 problems we cannot provide help

**Comment by overkill32** [comment]
Thank you @asbiin.  I have followed your advice to fix it manually and have learned the following:

The migration for 2020_02_18_211620_add_contact_field_label is failing on my instance.  This appears to fail silently when using ```php artisan monica:update --force```, so I have not noticed all my migrations failing since then.

I dropped tables contact_field_labels and contact_field_contact_field_label and then re-ran ```php artisan migrate```.   This failed with error message: 

```
SQLSTATE[42000]: Syntax error or access violation: 1059 Identifier name 'dukecontact_field_contact_field_label_contact_field_label_id_foreign' is too long (SQL: a  
  lter table `dukecontact_field_contact_field_label` add constraint `dukecontact_field_contact_field_label_contact_field_label_id_foreign` foreign key (`contact_fie  
  ld_label_id`) references `dukecontact_field_labels` (`id`) on delete cascade) 
```

(Note that DB_PREFIX=duke)

So, now I believe this to be related to issue #4751 where key names are too long if DB_PREFIX is non-zero.

**Comment by overkill32** [comment]
Adding another note for documentation of future searchers...

Since I have no real need for DB_PREFIX, I built a new instance without it and migrated my data over. Since I was unable to use the UI to export data, I did so with mysqldump directly with ```docker-compose exec db /usr/bin/mysqldump -u root --password=*** monica > monica.sql```.   Note that this included the failed migration 2020_02_18_211620_add_contact_field_label and missed everything since.

Upon dumping the sql file, I used sed to strip out the DB_PREFIX string ("duke") and then restore the data into the fresh install as follows:
```
cat monica.sql.Dec24 | sed 's/duke//g' |  docker-compose exec -T db /usr/bin/mysql -u root --password=*** monica
```

This was successful, but still included the failed migration.   I then logged into mysql directly and dropped tables referenced in the 2020_02_18_211620_add_contact_field_label migration (this cost me some minor data, but I didn't care enough to extract and insert separately).   After the tables were gone, the migration ran successfully and then caught up all migrations to present.

I am now current on 2.19.1 with my data.  I am no longer using DB_PREFIX in my .env.
